File: Go/Solutions/projeuler/Problem017.go
package euler

import (
	"fmt"
	"time"
)

/*
Problem017 answers the problem at : https://projecteuler.net/problem=17

* Problem 17:


	If the numbers 1 to 5 are written out in words: one, two, three, four, five, then there are 3 + 3 + 5 + 4 + 4 = 19 letters used in total.

	If all the numbers from 1 to 1000 (one thousand) inclusive were written out in words, how many letters would be used?

	NOTE: Do not count spaces or hyphens. For example, 342 (three hundred and forty-two) contains 23 letters and 115 (one hundred and fifteen) contains 20 letters. The use of "and" when writing out numbers is in compliance with British usage.


*/
func Problem017() {
	start := time.Now()
	ans := 0

	for i := 1; i < 1001; i++ {
		ans += len(convTill1000(i))
	}

	end := time.Now()

	fmt.Printf("\nAnswer to Problem 17 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())

}

File: Python/Problem017.py
import num2words
from time import time

''' 
Problem 17: https://projecteuler.net/problem=17

If all the numbers from 1 to 1000 (one thousand) inclusive were written out in words, how many letters would be used?

* NOTE: Do not count spaces or hyphens. For example, 342 (three hundred and forty-two) contains 23 letters and 115 (one hundred and fifteen) contains 20 letters. The use of "and" when writing out numbers is in compliance with British usage.
'''

start = time.time()
word = lambda k : num2words.num2words(k).replace(' ', '').replace('-', '')
p = ''

for i in range(1, 1001):
    p += word(i)
    
ans = len(p)

if __name__ == '__main__':
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")
